cli/internal/daemon/daemon.go

```diff
@@ -0,0 +1,325 @@
+package daemon
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"time"
+
+	"github.com/adrg/xdg"
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+	"github.com/nightlyone/lockfile"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/vercel/turborepo/cli/internal/config"
+	"github.com/vercel/turborepo/cli/internal/daemon/connector"
+	"github.com/vercel/turborepo/cli/internal/fs"
+	"github.com/vercel/turborepo/cli/internal/server"
+	"github.com/vercel/turborepo/cli/internal/signals"
+	"github.com/vercel/turborepo/cli/internal/util"
+	"google.golang.org/grpc"
+)
+
+// Command is the wrapper around the daemon command until we port fully to cobra
+type Command struct {
+	Config        *config.Config
+	UI            cli.Ui
+	SignalWatcher *signals.Watcher
+}
+
+// Run runs the daemon command
+func (c *Command) Run(args []string) int {
+	cmd := getCmd(c.Config, c.UI, c.SignalWatcher)
+	cmd.SetArgs(args)
+	err := cmd.Execute()
+	if err != nil {
+		return 1
+	}
+	return 0
+}
+
+// Help returns information about the `daemon` command
+func (c *Command) Help() string {
+	cmd := getCmd(c.Config, c.UI, c.SignalWatcher)
+	return util.HelpForCobraCmd(cmd)
+}
+
+// Synopsis of daemon command
+func (c *Command) Synopsis() string {
+	cmd := getCmd(c.Config, c.UI, c.SignalWatcher)
+	return cmd.Short
+}
+
+type daemon struct {
+	logger     hclog.Logger
+	repoRoot   fs.AbsolutePath
+	timeout    time.Duration
+	reqCh      chan struct{}
+	timedOutCh chan struct{}
+	ctx        context.Context
+	cancel     context.CancelFunc
+}
+
+func getRepoHash(repoRoot fs.AbsolutePath) string {
+	pathHash := sha256.Sum256([]byte(repoRoot.ToString()))
+	// We grab a substring of the hash because there is a 108-character limit on the length
+	// of a filepath for unix domain socket.
+	return hex.EncodeToString(pathHash[:])[:16]
+}
+
+func getDaemonFileRoot(repoRoot fs.AbsolutePath) fs.AbsolutePath {
+	tempDir := fs.GetTempDir("turbod")
+	hexHash := getRepoHash(repoRoot)
+	return tempDir.Join(hexHash)
+}
+
+func getLogFilePath(repoRoot fs.AbsolutePath) (fs.AbsolutePath, error) {
+	hexHash := getRepoHash(repoRoot)
+	base := repoRoot.Base()
+	logFilename := fmt.Sprintf("%v-%v.log", hexHash, base)
+
+	logsDirRaw, err := xdg.DataFile("logs")
```

Comment:
This is `~/Library/Application Support/logs` ... we should probably create our own directory to namespace things. `Turborepo/logs`, but lol who knows what path structure it needs.

Also, this API from XDG has side effects, so we should consider carefully when this gets called.

Reply:
What do you think about a policy of only ever using the "constants" (not actually constant) from `xdg`? I put a wrapper in `path.go` to get the `DataHome` field as an `AbsolutePath`. We can use our existing code to do any work we need to do off of that. That gives us better control of the side effects.

Reply:
I'm 100% on board with getting the base path out of `xdg` and then moving it into our path handling infrastructure as soon as possible. That will make our output far more predictable.